From: Egbert Eich <eich@suse.de>
Date: Sat Aug 22 10:51:45 2015 +0200
Subject: drm/i915: Use an interrupt save spinlock in intel_hpd_irq_handler()
Patch-mainline: N/A
References: bsc#942938

The IRQ save variant of spinlocks is used everywhere else for irq_lock.

Signed-off-by: Egbert Eich <eich@suse.de>
Signed-off-by: Egbert Eich <eich@suse.com>
---
 drivers/gpu/drm/i915/i915_irq.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)
diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index 391eb45..3eae834 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -991,11 +991,11 @@ static inline void intel_hpd_irq_handler(struct drm_device *dev,
 	drm_i915_private_t *dev_priv = dev->dev_private;
 	int i;
 	bool storm_detected = false;
+	unsigned long irqflags;
 
 	if (!hotplug_trigger)
 		return;
-
-	spin_lock(&dev_priv->irq_lock);
+	spin_lock_irqsave(&dev_priv->irq_lock, irqflags);
 	for (i = 1; i < HPD_NUM_PINS; i++) {
 
 		WARN(((hpd[i] & hotplug_trigger) &&
@@ -1027,7 +1027,7 @@ static inline void intel_hpd_irq_handler(struct drm_device *dev,
 
 	if (storm_detected)
 		dev_priv->display.hpd_irq_setup(dev);
-	spin_unlock(&dev_priv->irq_lock);
+	spin_unlock_irqrestore(&dev_priv->irq_lock, irqflags);
 
 	/*
 	 * Our hotplug handler can grab modeset locks (by calling down into the
